src/main/java/strategy/PurchaseHandler.java

```diff
@@ -0,0 +1,16 @@
+package strategy;
+
+import dao.StorageDao;
+
+public class PurchaseHandler implements FruitsAmountHandler {
+    private StorageDao storageDao;
+
+    public PurchaseHandler(StorageDao storageDao) {
+        this.storageDao = storageDao;
+    }
+
+    @Override
+    public void updateAmount(String name, Integer addNumber) {
+        storageDao.update(name, storageDao.get(name) - addNumber);

```

Comment:
it would be good to check if there is enough fruit to make a purchase in the storage